packages/@aws-cdk/aws-cloudformation/lib/pipeline-actions.ts

```diff
@@ -210,6 +210,10 @@ export abstract class PipelineCloudFormationDeployAction extends PipelineCloudFo
         this.role.addToPolicy(new iam.PolicyStatement().addAction('*').addAllResources());
       }
     }
+
+    props.stage.pipelineRole.addToPolicy(new iam.PolicyStatement()
```

Comment:
I don't reckon that's needed for all actions (you need this only to `CreateReplaceChangeSet`, but not to `ExecuteChangeSet`, for example). Factoring this here violates the principle of least privilege to me. Since this allows passing a role with permissions to do "everything" (via CloudFormation), I would argue we need this to be as tight as possible.

Reply:
It's not needed for all Actions. This code has been moved to `PipelineCloudFormationDeployAction`, of which all subclasses need it. `ExecuteChangeSet` extends a different class (`PipelineCloudFormationAction`), and thus does not have this permissions.

Reply:
> Add a comment explaining why this is needed

Done.